src/setup-dotnet.ts

```diff
@@ -116,4 +104,37 @@ function getVersionFromGlobalJson(globalJsonPath: string): string {
   return version;
 }
 
+function outputInstalledVersion(
+  installedVersions: (string | null)[],
+  globalJsonFileInput: string
+): void {
+  if (!installedVersions.length) {
+    core.info(`The 'dotnet-version' output will not be set.`);
+    return;
+  }
+
+  if (installedVersions.includes(null)) {
+    core.warning(
+      `Failed to output the installed version of .NET. The 'dotnet-version' output will not be set.`
+    );
+    return;
+  }
+
+  if (globalJsonFileInput) {
+    const versionToOutput = installedVersions.at(-1);
+    core.setOutput('dotnet-version', versionToOutput);
+    return;
+  }
+
+  const versionToOutput = semver.maxSatisfying(
+    installedVersions as string[],
```

Comment:
It's shouldn't crash as on the line 116 there is check for nulls in the `installedVersions` array. If it includes null (it may mean that dotnet-installer scripts changed their logs structure and we can't parse version out of them correctly), we send warning and return without setting `dotnet-version` output.